XMonad/Hooks/DynamicIcons.hs

```diff
@@ -95,29 +95,40 @@ dynamicIconsPP ic pp = getWorkspaceIcons ic <&> \ren -> pp{ ppRename = ppRename
 -- | Returns a function for 'ppRename' that augments workspaces with icons
 -- according to the provided 'IconConfig'.
 getWorkspaceIcons :: IconConfig -> X (String -> WindowSpace -> String)
-getWorkspaceIcons IconConfig{..} = fmt <$> getWorkspaceIcons' iconConfigIcons
+getWorkspaceIcons IconConfig{..} = fmt <$> iconFilterFunction iconConfigIcons
   where
     fmt icons s w = iconConfigFmt s (M.findWithDefault [] (S.tag w) icons)
 
+-- | Only use the focused window for each workspace to find icon
+getMasterIcon :: Query [String] -> X (M.Map WorkspaceId [String])
```

Comment:
why is this called `getMasterIcon`?  Wouldn't `getFocusedIcon` or something like that be more precise?